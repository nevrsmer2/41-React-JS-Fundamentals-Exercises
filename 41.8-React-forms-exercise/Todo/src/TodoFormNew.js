import React, { useState } from 'react';


const TodoFormNew = ({ addTodo }) => {
    const INITIAL_STATE = {
        todo: ""
    };

    const [formData, setFormData] = useState(INITIAL_STATE)

    const handleChange = e => {
        const { name, value } = e.target;
        setFormData(data => ({
            ...data,
            [name]: value
        }));
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        addTodo({ ...formData })
        setFormData(INITIAL_STATE)
    };

    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="todo">New Task: </label>
            <input
                id="todo"
                type="text"
                placeholder="New Todo"
                name="todo"
                value={formData.todo}
                onChange={handleChange}
            >
            </input>
            <br></br>
            <button>Add Task</button>
        </form>
    );
};

export default TodoFormNew;

