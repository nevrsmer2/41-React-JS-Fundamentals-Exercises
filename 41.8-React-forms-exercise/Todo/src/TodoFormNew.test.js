import { render } from "@testing-library/react";
import React from "react";
import TodoFormNew from "./TodoFormNew";

// Smike Test
it("TodoFormNew component renders without crashing", function () {
    render(<TodoFormNew />);
});

// Snapshot Test 
it("Current structure matches structure in snapshot", function () {
    const { asFragment } = render(<TodoFormNew />);
    expect(asFragment()).toMatchSnapshot();
});