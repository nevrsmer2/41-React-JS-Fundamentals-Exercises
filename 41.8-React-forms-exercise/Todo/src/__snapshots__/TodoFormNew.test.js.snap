// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current structure matches structure in snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="todo"
    >
      New Task: 
    </label>
    <input
      id="todo"
      name="todo"
      placeholder="New Todo"
      type="text"
      value=""
    />
    <br />
    <button>
      Add Task
    </button>
  </form>
</DocumentFragment>
`;
